### PR TITLE
Removed Loggers and added Application Insights

### PR DESCRIPTION
**Pull request by TarkusT**

### CODE REVIEW

File: application/Program.cs [hunk 1]
```diff
@@ -10,17 +14,30 @@ public class Program
 	{
 		public static void Main(string[] args)
 		{
-            using (var amz = new AmazonSynchronizer(new Configuration()))
+            var config = new Configuration();
+            var telemetry = new TelemetryClient();
+		    telemetry.InstrumentationKey = config.InstrumentationKey;
+		    telemetry.Context.Device.Id = Environment.MachineName;
+            telemetry.Context.Session.Id = Guid.NewGuid().ToString();
+		    telemetry.Context.Location.Ip = Dns.GetHostEntryAsync(Dns.GetHostName())
```
Comment (MicahZoltu) [inline application/Program.cs:22]: Hmm, doesn't `ApplicationInsights` automatically add all these for you?

Comment (TarkusT) [inline application/Program.cs:22]: I don't think so, it's not like the asp version.

https://azure.microsoft.com/en-us/documentation/articles/app-insights-windows-desktop/

File: application/Synchronization/AmazonSynchronizer.cs [hunk 2]
```diff
@@ -125,10 +139,14 @@ private void ExecuteAndWait(Action action, int delayInMs = 1000)
...
-                _logger.WriteEntry($"@ExecuteAndWait : {ex.GetLastErrorMessage()}", LoggingLevel.Error, _updatesCount);
+                _telemetryClient.TrackException(ex, new Dictionary<string, string>()
+                {
+                    {"Source", "ExecuteAndWait" },
+                    {"Update#" ,_updatesCount.ToString()}
+                });
             }
 
-            _logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
+            //_logger.WriteEntry($"Took : {_watch.ElapsedMilliseconds} Ms", LoggingLevel.Debug, _updatesCount);
```
Comment (MicahZoltu) [inline application/Synchronization/AmazonSynchronizer.cs:149]: There is a way to record timings in `ApplicationInsights`, I don't remember what the name of the method is but one of them takes a duration I believe.  Would be nice to get this recorded as such.

File: application/Synchronization/SynchronizerBase.cs [hunk 3]
```diff
@@ -154,6 +134,9 @@ public void Dispose(bool disposing)
             {
                 //cleanup
                 Stop();
+                _amazonClient = null;
+                _watch = null;
+                //_telemetryClient.Flush();
```
Comment (MicahZoltu) [inline application/Synchronization/SynchronizerBase.cs:139]: Why is this commented out?  If it is unnecessary, can it be removed instead?  If it is necessary, should it be uncommented?

Comment (TarkusT) [inline application/Synchronization/SynchronizerBase.cs:139]: I flush the client in the Program.cs

I will remove it from the Dispose

File: application/Synchronization/AmazonSynchronizer.cs (hunk 2 above)
Comment (TarkusT) [inline application/Synchronization/AmazonSynchronizer.cs:149]: I think it's done using TrackDependency or something like that.

I don't know if the entries have a timestamp.if so you don't need to record the timing.

**Pull request merged**